docs/topics/coroutines-basics.md

```diff
@@ -1,293 +1,565 @@
+> 
+{style="tip"}
+
+## Coroutine scope and structured concurrency
 
-### Structured concurrency
+When you run many coroutines in an application, you need a way to manage them as groups.
+Kotlin coroutines rely on a principle called _structured concurrency_ to provide this structure.
 
-Coroutines follow a principle of 
-**structured concurrency** which means that new coroutines can only be launched in a specific [CoroutineScope]
-which delimits the lifetime of the coroutine. The above example shows that [runBlocking] establishes the corresponding
-scope and that is why the previous example waits until `World!` is printed after a second's delay and only then exits.
+This principle groups coroutines into a hierarchy of parent and child tasks with linked lifecycles.
+Coroutines form a tree hierarchy where each coroutine has at most one parent and can have any number of children.
+A parent coroutine waits for its children to complete before it completes.
+If the parent coroutine fails or is canceled, all its child coroutines are canceled too.
+Keeping coroutines connected this way makes cancellation, error handling, and resource cleanup predictable and safe.
 
-In a real application, you will be launching a lot of coroutines. Structured concurrency ensures that they are not
-lost and do not leak. An outer scope cannot complete until all its children coroutines complete. 
-Structured concurrency also ensures that any errors in the code are properly reported and are never lost.  
+> A coroutine's lifecycle is the period from its creation until it completes, fails, or is canceled.
+>
+{style="tip"}
+
+To maintain structured concurrency, new coroutines can only be launched in a [`CoroutineScope`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/) that defines and manages their lifecycle.
+When you start a coroutine inside another coroutine, it automatically becomes a child of its parent scope.
+The parent coroutine's scope waits for all its children to finish before it completes.
+
+Calling a [coroutine builder function](#coroutine-builder-functions) such as `.launch()` on a `CoroutineScope` starts a child coroutine of the coroutine associated with that scope.
+Inside the builder's block, the [receiver](lambdas.md#function-literals-with-receiver) is a `CoroutineScope`, so any coroutines you launch there become its children.
+I don't want to be too explicit here. I want to explain this further down in the next section. 
 
-## Extract function refactoring
+While you can use the `withContext()` function to create a scope for your coroutines,
+you can use the [`coroutineScope()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/coroutine-scope.html) function when you want to create a new coroutine scope without changing the context.
+It executes the suspending block inline and suspends until the block and any coroutines launched in it complete.
```

Comment:
I like this phrasing better than previously. I'm still not sure if it's 100% correct because it _potentially_ suspends if the code inside it suspends. Saying "it waits until the block and coroutines completes" would be more correct, I believe.

(tagging @dkhalanskyjb here once more since the previous discussion got marked as "outdated" so not sure if that'll still show up in your reviews)